pymeasure/display/windows.py

```diff
@@ -430,16 +441,27 @@ def make_procedure(self):
                             " without a InputsWidget type")
         return self.inputs.get_procedure()
 
-    def new_curve(self, results, color=None, **kwargs):
+    def new_curve(self, wdg, results, color=None, **kwargs):
         if color is None:
             color = pg.intColor(self.browser.topLevelItemCount() % 8)
-        return self.plot_widget.new_curve(results, color=color, **kwargs)
+        return wdg.new_curve(results, color=color, **kwargs)
 
     def new_experiment(self, results, curve=None):
-        if curve is None:
-            curve = self.new_curve(results)
-        browser_item = BrowserItem(results, curve)
-        return Experiment(results, curve, browser_item)
+        if curve == None:
```

Comment:
To check whether `curve` equals `None`, a `curve is None` is a better comparison, as there always will be only a single `None` object (and for some specific, maybe very rare, cases the `==` comparison can result in unexpected results)

Reply:
OK, thanks for the tip. I will do that

Reply:
Done